### ISSUE TITLE
GoogleTest should not define -Werror or equivalent by default

### ISSUE THREAD

**Issue by heinrich5991** [google/googletest#1943]
As mentioned in the now closed https://github.com/google/googletest/issues/1373#issuecomment-355572722, it would probably be a good idea to not enable `-Werror` (or equivalently `/WX`). This can unnecessarily break the build for new compiler versions.

https://github.com/google/googletest/blob/b9347b31c338851879d0105f0fe32d09007f0433/googletest/cmake/internal_utils.cmake#L102

https://github.com/google/googletest/blob/b9347b31c338851879d0105f0fe32d09007f0433/googletest/cmake/internal_utils.cmake#L70

**Comment by Neumann-A** [comment]
+1

Just one of many examples:
`G:\vcpkg\buildtrees\gtest\src\062b60e006-05d61266d4\googletest\include\gtest/internal/gtest-port.h(2497,44):  error: 'chdir' is deprecated: The POSIX name for this item is deprecated. Instead, use the ISO C and C++ conformant name: _chdir. See online help for details. [-Werror,-Wdeprecated-declarations]
inline int ChDir(const char* dir) { return chdir(dir); }
                                           ^`

How about an option if we want gtest to mess with compiler flags?

**Comment by gennadiycivil** [comment]
Thank you very much for this report. The best way to approach this would be to create a proper PR and submit it for consideration.

**Comment by gennadiycivil** [comment]
As per https://github.com/google/googletest/blob/master/CONTRIBUTING.md CLA is required. If anyone in the community feels strongly that the change suggested in https://github.com/google/googletest/pull/1952 is important please take a look. 
Thank you

**Comment by heinrich5991** [comment]
Was this issue resolved? I can still see `-Werror` in that file.

**Comment by EricWF** [comment]
I disagree with the premise of this PR.

Google Test should define `-Werror` by default, and it should be testing against all supported toolchains, so that we can detect regression immediately.  This requires Google test to be proactive about testing bleeding edge configurations; but I think that is fair.

As I understand, our use of `-Werror` doesn't transitively effect users of the library or headers; only people who are building our library.

At most we should offer a option to disable `-Werror` during the build, otherwise I would prefer to enable errors by default.

**Comment by heinrich5991** [comment]
@EricWF I agree that Google should be testing the library with `-Werror` enabled in development.

The preferred way to use Google Test is to build it yourself (https://github.com/google/googletest/blob/b5f5c596a9915106c1ac36a3f89db4e0e49c07d1/googletest/README.md#incorporating-into-an-existing-cmake-project), this is what I did and this is where I hit the `-Werror` defined by Google Test. Thus it does transitively affect users of the library, unless I'm misunderstanding something.

In my opinion, the `-Werror` should only be defined when the official builds run, on CI and so on, so that these warnings are caught when developing. As new compilers or rather new compiler versions appear, this `-Werror` flag might otherwise stop old versions of Google Test from building.

**Comment by heinrich5991** [comment]
@gennadiycivil The issue was not resolved, could you please re-open? See e.g. #1982.

**Comment by bensuperpc** [comment]
Do you know how to turn off "-Werror" ?

**Comment by maryfranharper** [comment]
No unfortunately not. Sorry
